common/src/main/java/com/wynntils/screens/maps/SeaskipperMapScreen.java

```diff
@@ -0,0 +1,522 @@
+                (int) (renderX + renderedBorderXOffset), (int) (renderY + renderedBorderYOffset), (int) mapWidth, (int)
+                        mapHeight);
+
+        renderPois(poseStack, mouseX, mouseY);
+
+        renderCursor(
+                poseStack,
+                1.5f,
+                Managers.Feature.getFeatureInstance(GuildMapFeature.class)
+                        .pointerColor
+                        .get(),
+                Managers.Feature.getFeatureInstance(GuildMapFeature.class)
+                        .pointerType
+                        .get());
+
+        RenderUtils.disableScissor();
+
+        renderBackground(poseStack);
+
+        renderCoordinates(poseStack, mouseX, mouseY);
+
+        renderMapButtons(poseStack, mouseX, mouseY, partialTick);
+
+        renderHoveredSeaskipperDestination(poseStack);
+    }
+
+    private void renderPois(PoseStack poseStack, int mouseX, int mouseY) {
+        Vec3 playerLocation = McUtils.player().position();
+
+        for (Poi poi : seaskipperPois) {
+            SeaskipperPoi seaskipperPoi = (SeaskipperPoi) poi;
+
+            if (seaskipperPoi.isPlayerInside(playerLocation)) {
+                currentPoi = poi;
+                break;
+            }
+        }
+
+        if (renderAllDestinations) {
+            renderPois(
+                    seaskipperPois,
+                    poseStack,
+                    BoundingBox.centered(mapCenterX, mapCenterZ, width / currentZoom, height / currentZoom),
+                    1,
+                    mouseX,
+                    mouseY);
+        } else {
+            availableDestinations = new ArrayList<>();
+
+            for (SeaskipperDestinationItem destinationItem : destinations.keySet()) {
+                String destination = destinationItem.getDestination();
+
+                for (Poi poi : seaskipperPois) {
+                    if (poi.getName().equals(destination)) {
+                        availableDestinations.add(poi);
+                        break;
+                    }
+                }
+            }
+
+            availableDestinations.add(currentPoi);
+
+            renderPois(
+                    availableDestinations,
+                    poseStack,
+                    BoundingBox.centered(mapCenterX, mapCenterZ, width / currentZoom, height / currentZoom),
+                    1,
+                    mouseX,
+                    mouseY);
+        }
+    }
+
+    @Override
+    protected void renderPois(
+            List<Poi> pois,
+            PoseStack poseStack,
+            BoundingBox textureBoundingBox,
+            float poiScale,
+            int mouseX,
+            int mouseY) {
+        hovered = null;
+
+        List<Poi> filteredPois = getRenderedPois(pois, textureBoundingBox, poiScale, mouseX, mouseY);
+
+        if (renderRoutes) {
+            for (Poi poi : availableDestinations) {
+                if (!(poi instanceof SeaskipperPoi destination)) continue;
```

Comment:
You can make the `availableDestinations` a list of `SeaskipperPoi` so you don't have to deal with this

Reply:
I'm not sure if I can, `getRenderedPois` which is from `AbstractMapScreen` has to take in a List of `Poi`

Reply:
I am not sure if you have to have a real implementation for that in this case, or if you can just return an empty list since you handle poi rendering yourself.